### PR TITLE
TE-10560 Documentation page for Scalable application infrastructure 

### PR DESCRIPTION
**Pull request by yaroslav-spryker**
## PR Description

Documentation page for Scalable application infrastructure for P&S workers

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+`R` - total RAM available for worker pool.
+
+`N` - thread pool size.
+
+# Project enablement
+
+For enabling Scalable application infrastructure for P&S workers, please see “Spryker elastic computing” document.
+
+# Implementation details
+Module diagram
```
Comment (yaroslav-spryker) [inline docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md:91]: Please add an image below
image

File: docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+Class diagram
```
Comment (yaroslav-spryker) [inline docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md:94]: Please add an image below
image

File: docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+* `N` — thread pool size.
+
+* `w` — weight of queue.
+
+> :warning: In case of insufficient pool size notification must be sent.
+
+## RAM distribution
+
+To exclude runtime memory limit issues, the memory allocated to each worker must be also managed.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md:80]: Managed by what?

File: docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+Symbols:
+
+* `n` — number of workers for particular queue.
+
+* `N` — thread pool size.
+
+* `w` — weight of queue.
+
+> :warning: In case of insufficient pool size notification must be sent.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md:76]: Sent by what? Sent where?

File: docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+Setting `THREAD_POOL_SIZE` to more than `0` disables the `DEFAULT_MAX_QUEUE_WORKER` parameter.
+
+{% endinfo_block %}
+
+
+## Query priority
+
+In some cases, a smaller amount of messages may require more workers for processing. To balance such cases, use the `QUEUE_PRIORITY` configurable parameter.
+
+`QUEUE_PRIORITY` - real number configurable ratio per queue. If a queue has higher priority in processing, based on this ration, it gets more workers in proportion to the number of messages. The default value is `1`.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md:34]: How is it real?

File: docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+This document describes the methods of improving performance and efficiency of resource utilization by queue processing operation.
+
+## Thread pool size
+
+`THREAD_POOL_SIZE` defines the maximum number of parallel processes for all queues. It's default value is `0`, which means that scalable infrastructure is disabled.
+
+At the beginning of the processing, queues contain a different number of messages. To process all messages efficiently, processes need to be distributed among queues according to the number of messages per queue.
+
+> :info: `max_children` value must be reviewed in **php-fpm** server configuration according to configured `THREAD_POOL_SIZE`.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md:19]: What for?

File: docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+Performing compute-intensive operations with significant amount of data without proper resource management inevitably leads to inefficient computation resource utilization and increases processing time.
+
+This document describes the methods of improving performance and efficiency of resource utilization by queue processing operation.
+
+## Thread pool size
+
+`THREAD_POOL_SIZE` defines the maximum number of parallel processes for all queues. It's default value is `0`, which means that scalable infrastructure is disabled.
+
+At the beginning of the processing, queues contain a different number of messages. To process all messages efficiently, processes need to be distributed among queues according to the number of messages per queue.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/performance-guidelines/scalable-application-infrastructure.md:17]: So how do I make sure they are distributed?

**Review by qlmmlp — APPROVED** [review]

**Review by andriitserkovnyi — APPROVED** [review]

**Pull request merged**